src/responses/item-extra-info.ts

```diff
@@ -0,0 +1,102 @@
+import { Memoize } from 'typescript-memoize';
+import { makeReview } from '../models/review-builder';
+import { Review } from './page-elements';
+import { Metadata } from '@internetarchive/iaux-item-metadata';
+
+/**
+ * Extra info about the target item that is returned for
+ * the `item_details` page type.
+ */
+export class ItemExtraInfo {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  readonly rawResponse: Readonly<Record<string, any>>;
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  constructor(json: Record<string, any>) {
+    this.rawResponse = json;
+  }
+
+  /** Total size in bytes of this item (incl. its members) */
+  get item_size(): number | undefined {
+    return this.rawResponse.item_size;
+  }
+
+  /** How many files are within this item proper (not incl. its members) */
+  get files_count(): number | undefined {
+    return this.rawResponse.files_count;
+  }
+
+  /** How many views this item has received in the past month */
+  get month(): number | undefined {
+    return this.rawResponse.month;
+  }
+
+  /** How many views this item has received in the past week */
+  get week(): number | undefined {
+    return this.rawResponse.week;
+  }
+
+  /** How many total views this item has received (all-time) */
+  get downloads(): number | undefined {
+    return this.rawResponse.downloads;
+  }
+
+  /** How many users have favorited this item */
+  get num_favorites(): number | undefined {
+    return this.rawResponse.num_favorites;
+  }
+
+  /** Additional message to appear in the page title for this item, if any */
+  get title_message(): string | null | undefined {
+    return this.rawResponse.title_message;
+  }
+
+  /** The primary parent collection that contains this one as a member, if any */
+  get primary_collection(): string | null | undefined {
+    return this.rawResponse.primary_collection;
+  }
+
+  @Memoize() get reviews_data(): Review[] | undefined {
+    return this.rawResponse.reviews_data.map(makeReview);
+  }
```

Comment:
Nice!

Q: Is it actually possible for this to return `undefined`? And if so, do we need to add optional chaining anywhere here?

Reply:
Ah good point. I made this always return an array.